framework/helpers/BaseInflector.php

```diff
@@ -373,7 +373,7 @@ public static function camel2words($name, $ucwords = true)
             '-',
             '_',
             '.',
-        ], ' ', preg_replace('/(\p{Lu})/u', ' \0', $name))), self::encoding());
+        ], ' ', preg_replace('/(?<!\p{Lu})(\p{Lu})|(\p{Lu})(?=\p{Ll})/u', ' \0', $name))), self::encoding());
```

Comment:
what is the or branch of the regex for? This part: `(\p{Lu})(?=\p{Ll})`

Original regex (in 2.0.15 was `/(?<![A-Z])[A-Z]/`), making that unicode aware would be just: `/(?<!\p{Lu})\p{Lu}/`

Reply:
@cebe It looks for capital letters that are followed by a lowercase letter. Without it, `ITest` would be returned as `Itest` (as it did pre-2.0.16), but that would break this new test added in 2.0.16:

https://github.com/yiisoft/yii2/blob/ed64d6588630f9a1df92041fd5ca4087e786f3fb/tests/framework/helpers/InflectorTest.php#L108

(`Іце Дійсно Так!` would be returned instead of `І Це Дійсно Так!`.)

Reply:
Got it, thanks.